### PR TITLE
Temporary Strings to use StringBuffer.

### PR DESCRIPTION
**Pull request by simosillankorva**

### CODE REVIEW

**Comment by simosillankorva** [comment]
rebuild

**Comment by simosillankorva** [comment]
rebuild

**Comment by simosillankorva** [comment]
This task is panding for review. @anttiylitokola @kuggenhoffen

File: source/m2mbase.cpp [hunk 1]
```diff
@@ -443,3 +444,73 @@ void M2MBase::execute_value_updated(const String& name)
         _value_updated_callback(name.c_str());
     }
 }
+
+bool M2MBase::build_path(StringBuffer<MAX_PAHTH_SIZE> &buffer, const char *s1, uint16_t i1, const char *s2, uint16_t i2)
+{
+
+    if(!buffer.ensure_space(strlen(s1) + strlen(s2) + 10 + 3 + 1)){
```
Comment (anttiylitokola) [inline source/m2mbase.cpp]: What are these magic numbers? Can those be defined somewhere.

**Comment by simosillankorva** [comment]
rebuild

**Comment by simosillankorva** [comment]
Issues found by @anttiylitokola and @kuggenhoffen should now be fixed, you could check the fixes before I merge this.

**Comment by kuggenhoffen** [comment]
+1

**Pull request merged**